pkg/storage/pvc/pvc.go

```diff
@@ -116,13 +117,36 @@ func (d *driver) checkPVC(cr *imageregistryv1.Config, claim *corev1.PersistentVo
 		}
 	}
 
-	for _, claimMode := range claim.Spec.AccessModes {
+	// Check what access modes are available.
+
+	// We allow using RWO PV backend, but it has some limitations:
+	// 1. Image registry rollout strategy must be set to Recreate (default is RollingUpdate).
+	// 2. It's not possible to use more than 1 replica of the image registry.
+
+	// RWX backends are accepted with no additional conditions.
+	RWOModeEnabled := false
+	for _, claimMode := range claim.Status.AccessModes {
 		if claimMode == corev1.ReadWriteMany {
 			return nil
 		}
+		if claimMode == corev1.ReadWriteOnce {
+			RWOModeEnabled = true
+		}
+	}
+
+	if RWOModeEnabled {
+		if cr.Spec.Replicas > 1 {
+			return fmt.Errorf("Cannot use %s access mode with more than one replica of the image registry", corev1.ReadWriteOnce)
+		}
+
+		if cr.Spec.RolloutStrategy != string(appsv1.RecreateDeploymentStrategyType) {
+			return fmt.Errorf("Cannot use %s access mode with %s rollout strategy", corev1.ReadWriteOnce, cr.Spec.RolloutStrategy)
```

Comment:
nit - start with lowercase
```suggestion
			return fmt.Errorf("cannot use %s access mode with %s rollout strategy", corev1.ReadWriteOnce, cr.Spec.RolloutStrategy)
```

Reply:
done